nengo_gui/components/value.py

```diff
@@ -59,6 +59,9 @@ def update_client(self, client):
         # while there is data that should be sent to the client
         while len(self.data) > 0:
             item = self.data.popleft()
+            while len(self.data) > 0:
```

Comment:
> Hmm, that's a lot less aggressive than the current code, in that it doesn't immediately send the data that has queued up while it's been busy putting together this list of data. But I could do some testing to see if that less complete approach also works.

So I did a bit of testing on a few machines, and it looks like replacing the outer ```while``` with an ```if``` seems okay.  It definitely lags a little more, as I was suggesting, but it turns out to only be around 10 timesteps more in the worst case I could find, which is completely unnoticeable to the user.  So I'm happy getting rid of the outer while loop just to make the code look clearer.